Lib/pathlib.py

```diff
@@ -183,19 +183,18 @@ def resolve(self, path, strict=False):
             if strict:
                 return self._ext_to_normal(_getfinalpathname(s))
             else:
+                tail_parts = []  # End of the path after the first one not found
                 while True:
                     try:
                         s = self._ext_to_normal(_getfinalpathname(s))
                     except FileNotFoundError:
                         previous_s = s
-                        s = os.path.dirname(s)
+                        s, tail = os.path.split(s)
```

Comment:
@pitrou I did some tests and I'm really not sure what the problem is with UNC paths... I think if there is a problem, it won't be introduced by this patch anyway as this code already use `os.path.dirname`, so it's probably irrelevant?

Since the goal is to backport that fix, I thought it would be best to keep it as simple as possible.